code/modules/mob/interactive.dm

```diff
@@ -726,6 +723,8 @@
 					TARGET = pick(target_filter(favouredObjIn(urange(MAX_RANGE_FIND,src,1))))
 				if(46 to 50)
 					TARGET = pick(target_filter(oview(MIN_RANGE_FIND,src)))
+		else if(isTraitor && traitorTarget)
+			TARGET = traitorTarget
```

Comment:
This just makes the SPNC stall forever if the traitorTarget is more than MAX_RANGE_FIND distance away, because it can't path to it.